src/main/java/net/dv8tion/jda/core/EmbedBuilder.java

```diff
@@ -299,14 +306,26 @@ public EmbedBuilder setFooter(String text, String iconUrl)
         {
             if (text != null && text.length() > TEXT_MAX_LENGTH)
                 throw new IllegalArgumentException("Text cannot be longer than " + TEXT_MAX_LENGTH + " characters.");
-            httpsCheck(iconUrl);
+            urlCheck(iconUrl);
             this.footer = new MessageEmbed.Footer(text, iconUrl, null);
         }
         return this;
     }
     
     /**
+     * Adds a Field to the embed
+     * 
+     * @param field the field object to add
+     * @return the builder after the field has been added
+     */
+    public EmbedBuilder addField(MessageEmbed.Field field)
+    {
+        return addField(field.getName(), field.getValue(), field.isInline());
```

Comment:
It could be (I'll put that in next time I'm updating this stuff), although the end result is the same: a NPE when someone tried to make `field` null (difference in description and internal cleanliness)